core/src/main/java/com/linecorp/armeria/common/stream/StreamWriter.java

```diff
@@ -132,8 +133,15 @@ default boolean tryWrite(Supplier<? extends T> o) {
     void close(Throwable cause);
 
     /**
-     * Writes the given object and closes the stream successfully.
+     * Writes the given object and closes the stream successfully. The written object will be transferred to
+     * the {@link Subscriber}.
+     *
+     * @throws ClosedPublisherException if the stream was already closed.
+     * @see <a href="#reference-counted">Life cycle of reference-counted objects</a>
+     *
+     * @deprecated Use {@link #tryWrite(Object)} and {@link #close()}.
      */
+    @Deprecated
     default void close(T obj) {
         write(obj);
         close();
```

Comment:
Don't we need to update this as well?
```java
    try {
        write(obj);
    } finally {
        close();
    }
```

Reply:
Just wanted to preserve the behavior, but I guess you're right.